### ISSUE TITLE
Platform selection

### ISSUE THREAD

**Issue by amwalters** [diegoveloper/flutter_keyboard_actions#34]
Platform selection does not appear to be working, and I cannot see where you are checking which platform was selected.

I was able to fix it by adding _isAvailable to _focusChanged as below

```
  _focusChanged(bool showBar) {
    if (showBar && !_isShowing && _isAvailable) {
      _insertOverlay();
    } else if (!showBar && _isShowing && _isAvailable) {
      _removeOverlay();
    } else if (showBar && _isShowing && _isAvailable) {
      _overlayEntry.markNeedsBuild();
    }

    WidgetsBinding.instance.addPostFrameCallback((_) {
      _updateOffset();
    });
  }
```

**Comment by diegoveloper** [comment]
Could you add the code to reproduce the issue ?  Could you try replacing :

      TargetPlatform.iOS by   Platform.isIOS;
and  TargetPlatform.android by Platform.isAndroid;

Let me know if it works

**Comment by amwalters** [comment]
That does not seem to fix it.  Your _isAvailable function works perfectly and will return false on my android if KeyboardActionsPlatform.IOS is set.  However it still will render the done button.

Where are you calling _isAvailable?

There is only one reference to it is under _updateOffset.  Don't you need to call it before building the overlay?

Code I'm using:
```
FormKeyboardActions(
                autoScroll: true,
                child:Builder(
                  builder: (BuildContext context) {
                    /// Creates the [KeyboardActionsConfig] to hook up the fields
                    /// and their focus nodes to our [FormKeyboardActions].
                    /// Only used in iOS to provide keyboard dismissal
                    KeyboardActionsConfig _buildConfig(BuildContext context) {
                      return KeyboardActionsConfig(
                        keyboardActionsPlatform: KeyboardActionsPlatform.IOS,     //not working disabled for android in init
                        keyboardBarColor: Colors.grey[200],
                        nextFocus: false,                                         //disable up/down arrows
                        actions: [
                          KeyboardAction(focusNode: _nodeWeight),
                          KeyboardAction(focusNode: bupivacaine.adminNode,),
                          KeyboardAction(focusNode: ropivacaine.adminNode,),
                          KeyboardAction(focusNode: mepivacaine.adminNode,),
                          KeyboardAction(focusNode: lidocaine.adminNode,),
                          KeyboardAction(focusNode: chloroprocaine.adminNode,),
                        ],
                      );
                    }

                    if (Platform.isIOS) {
                      FormKeyboardActions.setKeyboardActions(context, _buildConfig(context));
                    }
                    FormKeyboardActions.setKeyboardActions(context, _buildConfig(context));

                    return SingleChildScrollView(
                        controller: drugCardController,
                        //physics: AlwaysScrollableScrollPhysics(), //Scroll from here so we include the bottom buttons
                        child:Column(
                          children: <Widget>[
                            ListView.builder(
                                physics: NeverScrollableScrollPhysics(), //Don't scroll this list, were scrolling from above
                                shrinkWrap: true,
                                itemCount: drugList.length,
                                itemBuilder: (context, index) {
                                  final drug = drugList[index];

                                  return Dismissible(
                                    key: ObjectKey(drug.name),
                                    onDismissed: (direction) {
                                      setState(() {
                                        drugList.removeAt(index);
                                        drug.adminController.clear();
                                        _calculate();
                                      });
                                    },
                                    child: cardDrugRemaining(drug),
                                  );
                                }
                            ),
                            Padding(
                              padding: EdgeInsets.only(bottom:10, left:30, right:30),
                              child:Row(
                                mainAxisSize: MainAxisSize.max,
                                mainAxisAlignment: MainAxisAlignment.spaceAround,
                                crossAxisAlignment: CrossAxisAlignment.center,
                                children: <Widget>[
                                  Expanded(
                                    flex:1,
                                    child:Visibility(
                                        visible: drugList.length != 5,
                                        child: Container(
                                          margin: EdgeInsets.only(right: 10),
                                          child:MaterialButton(
                                            color: MyColors.PrimaryColor1inactive,
                                            child: Text('Show All Drugs'),
                                            onPressed: () {
                                              showAllCards();
                                            },
                                          ),
                                        )
                                    ),
                                  ),
                                  Expanded(
                                      flex:1,
                                      child: Container(
                                        margin: EdgeInsets.only(left: 10),
                                        child:MaterialButton(
                                          color: MyColors.PrimaryColor1inactive,
                                          child: Text('Clear Admin'),
                                          onPressed: () {
                                            drugList.forEach((drug)=> drug.adminController.clear());
                                          },
                                        ),
                                      )
                                  ),
                                ],
                              ),
                            ),
                          ],
                        )
                    );
                  },
                ),
              ),
```

**Comment by diegoveloper** [comment]
You can send a pull request

**Comment by amwalters** [comment]
Sadly I'm not that adept with github yet.  I am working around it using your base files and wrapping the initialization with an OS checker.

if (Platform.isIOS) {
   FormKeyboardActions.setKeyboardActions(context, _buildConfig(context));
}

**Comment by adolfdsilva** [comment]
Done button still shows for android, even after using 

`keyboardActionsPlatform: KeyboardActionsPlatform.IOS `

in  `KeyboardActionsConfig`

**Comment by diegoveloper** [comment]
Fixed on 3.1.2 version
